mava/components/jax/building/data_server.py

```diff
@@ -137,10 +280,22 @@ def table(
         Returns:
             _description_
         """
+        if not hasattr(builder.store, "sampler_fn"):
+            builder.store.sampler_fn = reverb.selectors.Uniform
+            warnings.warn(
+                "Sampler has not been expicitly chosen - defaults to uniform sampler."
+            )
+
+        if not hasattr(builder.store, "remover_fn"):
+            builder.store.remover_fn = reverb.selectors.Fifo
+            warnings.warn(
+                "Remover has not been expicitly chosen - defaults to FIFO remover."
+            )
+
```

Comment:
Not sure how I feel about defaulting to something here. If you look at the `rate_limiter_fn`, it doesn't have a default. It requires that a `RateLimiter` component be provided.

I think there should rather be checks here which throw an exception, since the user defined a data server component without providing a remover or a sampler component.
Whatever conclusion we go with, the same should be applied to the `rate_limiter`

Reply:
I have removed the warnings - I agree an exception should be raised - not sure where we should do this. Maybe in the same place where my warnings were.

Reply:
exceptions have been added.